src/coreclr/vm/greenthreads.cpp

```diff
@@ -54,6 +54,9 @@ extern "C" uintptr_t AllocateMoreStackHelper(int argumentStackSize, void* stackP
     uint8_t *newArgsLocation;
     if (argumentStackSize < 0)
     {
+        assert(t_greenThread.inGreenThread);
+        t_greenThread.inGreenThread = false;
```

Comment:
This pattern seems good for catching issues.  Perhaps use it every single time?  (even down below at line 231 which is 20 lines after the __debugbreak check, just in case, and again there when setting it back because the call could do something strange)